code/game/machinery/autolathe/artist_bench.dm

```diff
@@ -320,8 +320,14 @@
 
 	var/obj/artwork = choose_full_art(ins_used, user)
 	var/datum/design/art
-	if(isobj(artwork))
+	if (isobj(artwork))
 		art = new()
+		var/setname = sanitizeSafe(input(user,"Name your creation. Keep empty for a random name.","Set Name",""), MAX_NAME_LEN)
+		if (setname)
+			artwork.name = setname
+		var/setdesc = sanitizeSafe(input(user,"Describe your creation. Keep empty for a random description.","Set Description",""), MAX_DESC_LEN)
+		if (setdesc)
+			artwork.desc = setdesc
 		randomize_materialas(artwork)
```

Comment:
wouldnt this overight the ones you put in?

Reply:
nope. its there so that if the person keeps the name/desc fields empty, only then does it keep the randomised name and description. else if there's something that the user puts in those fields, its overwrites the randomgen name and puts what the player wants